src/main/java/io/IO.java

```diff
@@ -0,0 +1,172 @@
+package io;
+
+import task.Task;
+import task.TaskList;
+
+/*
+* Use this class for INPUT and OUTPUT (IO) related stuff
+* Got some inspiration from Contacts (Week 4)
+*/
+public final class IO {
+    /**
+     * ==============================================================
+     * This section will cover Inputs,
+     * Input Validations,
+     * and Input Processing Methods
+     * ==============================================================
+     */
+    public static final String COMMAND_HELP = "help";
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+
+    /** Add task commands. Sample usage as below. not delimited by spaces.
+     *  todo: "todo task_name"
+     *  deadline: "deadline task_name /by due_date"
+     *  event: "event task_name /from start_time /to end_time"
+     */
+    public static final String COMMAND_TASK_TODO = "todo";
+    public static final String COMMAND_TASK_DEADLINE = "deadline";
+    public static final String COMMAND_TASK_DEADLINE_DELIMITER = "/by";
+    public static final String COMMAND_TASK_DEADLINE_DELIMITER_REGEX =
+            "\\s+\\" + COMMAND_TASK_DEADLINE_DELIMITER + "\\s+";
+    public static final String COMMAND_TASK_EVENT = "event";
+    public static final String COMMAND_TASK_EVENT_DELIMITER1 = "/from";
+    public static final String COMMAND_TASK_EVENT_DELIMITER2 = "/to";
+    public static final String COMMAND_TASK_EVENT_DELIMITER_REGEX =
+            "\\s+\\" + COMMAND_TASK_EVENT_DELIMITER1 +
+            "\\s+|\\s+\\" + COMMAND_TASK_EVENT_DELIMITER2 + "\\s+";
+
+    /**
+     * Split input into 1st arg and subsequent line.
+     * @param inputLine A line of input direct from user
+     */
+    public static String[] splitCommandAndArgs(String inputLine) {
+        // Trim trailing and beginning whitespace, then split once. Output is [command, args].
+        final String[] split = inputLine.trim().split("\\s+", 2);
+        // Else case: no parameters
```

Comment:
```suggestion
        // Edge case: no parameters
```
But you can also consider raising an exception